File: selenium-framework/src/test/java/com/selenium/testng/context/ExecutionContext.java
package com.selenium.testng.context;

import java.text.SimpleDateFormat;
import java.util.Date;

public final class ExecutionContext {

	private static final String RUN_ID = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());

	public static String getRunId() {
		return RUN_ID;
	}

	public static String getLogPath() {

		return System.getProperty("user.dir") + "/logs/automation_" + RUN_ID + ".log";
	}

	public static String getReportPath() {

		return System.getProperty("user.dir") + "/reports/ExtentReport_" + RUN_ID + ".html";
	}

	public static String getScreenshotPath(String testName) {

		return System.getProperty("user.dir") + "/screenshots/" + testName + "_" + RUN_ID + ".png";
	}
}

File: src/framework/context/execution-context.ts
import path from 'path';

export class ExecutionContext {
  static getRunId(): string {
    return new Date().toISOString().replace(/[:.]/g, '-');
  }

  static getReportPath(): string {
    return path.resolve(process.cwd(), 'test-results', 'report', `report-${this.getRunId()}.html`);
  }

  static getLogPath(): string {
    return path.resolve(process.cwd(), 'test-results', 'logs', `framework-${this.getRunId()}.log`);
  }

  static getScreenshotPath(testName: string): string {
    return path.resolve(process.cwd(), 'test-results', 'screenshots', `${testName}-${this.getRunId()}.png`);
  }
}
